textrepo-app/src/main/java/nl/knaw/huc/db/DocumentFilesDao.java

```diff
@@ -40,12 +40,20 @@
 
   @SqlQuery("select id, type_id " +
       "from documents_files as df left join files as f on f.id = df.file_id " +
-      "where df.document_id = :docId limit :limit offset :offset")
+      "where df.document_id = :docId and (:typeId is null or f.type_id = :typeId) " +
+      "limit :limit offset :offset")
   @RegisterConstructorMapper(TextRepoFile.class)
-  List<TextRepoFile> findFilesByDocumentId(@Bind("docId") UUID docId, @BindBean PageParams pageParams);
+  List<TextRepoFile> findFilesByDocumentAndTypeId(
+      @Bind("docId") UUID docId,
+      @Bind("typeId") Short typeId,
+      @BindBean PageParams pageParams
+  );
 
   @SqlQuery("select count(*)" +
       "from documents_files as df left join files as f on f.id = df.file_id " +
-      "where df.document_id = ?")
-  long countByDocumentId(UUID docId);
+      "where df.document_id = :docId and (:typeId is null or f.type_id = :typeId)")
+  long countByDocumentAndTypeId(
+      @Bind("docId") UUID docId,
+      @Bind("typeId") Short typeId
```

Comment:
`@Nullable` erbij omdat deze echt ook bedoeld `null` mag zijn?

Reply:
Gefixt in 00c1434